keps/sig-apps/3998-job-success-completion-policy/README.md

```diff
@@ -0,0 +1,1100 @@
+
+<!--
+Even if applying deprecation policies, they may still surprise some users.
+-->
+
+No.
+
+### Monitoring Requirements
+
+<!--
+This section must be completed when targeting beta to a release.
+
+For GA, this section is required: approvers should be able to confirm the
+previous answers based on experience in the field.
+-->
+
+###### How can an operator determine if the feature is in use by workloads?
+
+<!--
+Ideally, this should be a metric. Operations against the Kubernetes API (e.g.,
+checking if there are objects with field X set) may be a last resort. Avoid
+logs or events for this purpose.
+-->
+
+`job_finished_total`: the new `JobSuccessfulPolicy` and `Completions` values for the `reason` label. 
+  - `JobSuccessfulPolicy` indicates a job is declared as succeeded because the job meets `.spec.successfulPolicy`.
+  - `Completions` indicates a job is declared as succeeded because the job mees `spec.completions`.
+
+###### How can someone using this feature know that it is working for their instance?
+
+<!--
+For instance, if this is a pod-related feature, it should be possible to determine if the feature is functioning properly
+for each individual pod.
+Pick one more of these and delete the rest.
+Please describe all items visible to end users below with sufficient detail so that they can verify correct enablement
+and operation of this feature.
+Recall that end users cannot usually observe component logs or access metrics.
+-->
+
+- [x] Job API .status
+  - The Job controller will add a condition with `JobSuccessfulPolicy` reason to `conditions`. 
+
+###### What are the reasonable SLOs (Service Level Objectives) for the enhancement?
+
+<!--
+This is your opportunity to define what "normal" quality of service looks like
+for a feature.
+
+It's impossible to provide comprehensive guidance, but at the very
+high level (needs more precise definitions) those may be things like:
+  - per-day percentage of API calls finishing with 5XX errors <= 1%
+  - 99% percentile over day of absolute value from (job creation time minus expected
+    job creation time) for cron job <= 10%
+  - 99.9% of /health requests per day finish with 200 code
+
+These goals will help you determine what you need to measure (SLIs) in the next
+question.
+-->
+
+###### What are the SLIs (Service Level Indicators) an operator can use to determine the health of the service?
+
+<!--
+Pick one more of these and delete the rest.
+-->
+
+- [x] Metrics
+  - Metric name: `job_sync_duration_seconds` (existing): can be used to see how much the
+    feature enablement increases the time spent in the sync job
+  - Components exposing the metric:
+
+###### Are there any missing metrics that would be useful to have to improve observability of this feature?
+
+<!--
+Describe the metrics themselves and the reasons why they weren't added (e.g., cost,
+implementation difficulties, etc.).
+-->
+
+No.
```

Comment:
Should we consider adding a metric for this feature? PodFailurePolicy has a metric so it may make sense to add one for success?

Reply:
I will add new reasons to `job_finished_total` for JobSuccessfulPolicy. I believe it is enough. What do you think?

> `job_finished_total`: the new `JobSuccessfulPolicy` and `Completions` values for the `reason` label. 
>   - `JobSuccessfulPolicy` indicates a job is declared as succeeded because the job meets `.spec.successfulPolicy`.
>   - `Completions` indicates a job is declared as succeeded because the job mees `spec.completions`.

Reply:
Sorry if I missed that. I think that sounds good to me.

Reply:
I mentioned that in `How can an operator determine if the feature is in use by workloads?` section!